bundles/org.openhab.binding.entsoe/src/main/java/org/openhab/binding/entsoe/internal/client/Client.java

```diff
@@ -0,0 +1,174 @@
+/**
+ * Copyright (c) 2010-2024 Contributors to the openHAB project
+ *
+ * See the NOTICE file(s) distributed with this work for additional
+ * information.
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0
+ *
+ * SPDX-License-Identifier: EPL-2.0
+ */
+package org.openhab.binding.entsoe.internal.client;
+
+import java.io.IOException;
+import java.io.StringReader;
+import java.time.Duration;
+import java.time.Instant;
+import java.time.ZoneId;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.LinkedHashMap;
+import java.util.Map;
+
+import javax.xml.parsers.DocumentBuilder;
+import javax.xml.parsers.DocumentBuilderFactory;
+import javax.xml.parsers.ParserConfigurationException;
+
+import org.eclipse.jdt.annotation.NonNullByDefault;
+import org.openhab.binding.entsoe.internal.exception.EntsoeConfigurationException;
+import org.openhab.binding.entsoe.internal.exception.EntsoeResponseException;
+import org.openhab.core.io.net.http.HttpUtil;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.w3c.dom.Document;
+import org.w3c.dom.Element;
+import org.w3c.dom.Node;
+import org.w3c.dom.NodeList;
+import org.xml.sax.InputSource;
+import org.xml.sax.SAXException;
+
+/**
+ * @author Miika Jukka - Initial contribution
+ * @author Jørgen Melhus - Contribution
+ */
+@NonNullByDefault
+public class Client {
+    private final Logger logger = LoggerFactory.getLogger(Client.class);
+
+    private DocumentBuilderFactory documentBuilderFactory = DocumentBuilderFactory.newInstance();
+
+    public Map<Instant, EntsoeTimeSerie> doGetRequest(Request request, int timeout, String configResolution)
+            throws EntsoeResponseException, EntsoeConfigurationException {
+        try {
+            logger.debug("Sending GET request with parameters: {}", request);
+            String url = request.toUrl();
+            String responseText = HttpUtil.executeUrl("GET", url, timeout);
+            if (responseText == null) {
+                throw new EntsoeResponseException("Request failed");
+            }
+            logger.trace("Response: {}", responseText);
+            return parseXmlResponse(responseText, configResolution);
+        } catch (IOException e) {
+            if (e.getMessage().contains("Authentication challenge without WWW-Authenticate header")) {
```

Comment:
This (still) generates a compiler warning.
```suggestion
            String message = e.getMessage();
            if (message != null && message.contains("Authentication challenge without WWW-Authenticate header")) {
```